dspace-oai/src/main/java/org/dspace/xoai/util/ItemUtils.java

```diff
@@ -318,4 +347,55 @@ public static Metadata retrieveMetadata (Context context, Item item) {
         
         return metadata;
     }
-}
+    
+	/**
+	 * Method to return a default value text to identify access rights:
+	 * 'open access','embargoed access','restricted access','metadata only access'
+	 *
+	 * NOTE: embargoed access contains also embargo end date in the form "embargoed access|||yyyy-MM-dd"
+	 *
+	 * @param rps
+	 * @return
+	 */
+	public static String getAccessRightsValue(List<ResourcePolicy> rps) {
+		Date now = new Date();
+		Date embargoEndDate = null;
+		boolean openAccess = false;
+		boolean groupRestricted = false;
+		boolean withEmbargo = false;
+
+		if (rps != null) {
+			for (ResourcePolicy rp : rps) {
+				if (Group.ANONYMOUS.equals(rp.getGroup().getName())) {
+					if (resourcePolicyService.isDateValid(rp)) {
+						openAccess = true;
+					} else if (rp.getStartDate() != null && rp.getStartDate().after(now)) {
+						withEmbargo = true;
+						embargoEndDate = rp.getStartDate();
+					}
+				} else if (!Group.ADMIN.equals(rp.getGroup().getName())) {
```

Comment:
It can throw a NullPointerException, rp.getGroup() could be null.

Reply:
Ok, noted.